generic/tests/nfs_corrupt.py

```diff
@@ -71,6 +74,20 @@ def setup(self, force_start=False):
 
         :param force_start: Whether to make NFS service start anyway.
         """
+        error.context("Checking available space to export")
+        tmpdir = os.path.split(self.nfs_dir)[0]
+        out = utils.system_output("df %s" % tmpdir,
+                                  ignore_status=True).split()[-3]
+
+        available_size = float(utils_misc.normalize_data_size(out,
+                                                              order_magnitude="M"))
+        required_size = float(utils_misc.normalize_data_size(self.required_size,
+                                                             order_magnitude="M"))
+        logging.debug("Space available: %fM, space needed: %fM"
```

Comment:
better to add this info in raise exceptions.TestError("Not enough free space to setup NFS")  to show more details of error info.

Reply:
@luckyh `utils_misc.normalize_data_size()` should probably return a float value.